kt/godot-runtime/src/main/kotlin/godot/runtime/Registration.kt

```diff
@@ -657,8 +657,9 @@ class ClassRegistry {
     fun <T : KtObject> registerClass(name: String, superClass: String, isTool: Boolean = false, cb: ClassBuilderDsl<T>.() -> Unit) {
         val builder = ClassBuilderDsl<T>(name, superClass)
         builder.cb()
-        TypeManager.registerUserType(name)
-        registerClass(builder.build())
+        val clazz = builder.build()
+        TypeManager.registerUserType(name) { voidPtr -> clazz.new(voidPtr, 0) }
```

Comment:
I'm assuming all user types has a zero-arg constructor here. I don't know if that's a reasonable assumption or what to do if it doesn't hold.